### PR TITLE
Add simple file server to demo-apps.

### PR DESCRIPTION
**Pull request by boaks**
Mainly intended for demonstrating LWM2M firmware update.

Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+					response.getOptions().setContentFormat(accept);
+					exchange.respond(response);
+				} else {
+					LOG.log(Level.WARNING, "File {0} could not be read in!", file.getAbsolutePath());
+					exchange.respond(CoAP.ResponseCode.INTERNAL_SERVER_ERROR);
+				}
+			} catch (IOException ex) {
+				LOG.log(Level.WARNING, "File {0}: {1}", new Object[] { file.getAbsolutePath(), ex });
+				exchange.respond(CoAP.ResponseCode.INTERNAL_SERVER_ERROR);
+			} catch (OutOfMemoryError error) {
```
Comment (sophokles73) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: When do we expect this to happen?

File: demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+		}
+	}
+
+	/*
+	 * Constructor for a new simple file server. Here, the resources of the
+	 * server are initialized.
+	 */
+	public SimpleFileServer() throws SocketException {
+
+		add(new FileResource("data", new File("data")));
```
Comment (sophokles73) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: should probably use a system property or command line arg for the root path, shouldn't it?

Comment (boaks) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: Command line arguments (and defaults) are a possibility.
But though I'm using a similar approach now for months, nobody asked me to change the file nor coap  path. Therefore to keep it really simple, I compiled in the values.

File: demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java (hunk 1 above)
Comment (boaks) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: IOException of OutOfMemoryError?
It's easy, I just want to have some logging and defined responses in the "exceptional" cases even if they never occur. But no problem we can remove one or both

File: demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java (hunk 2 above)
Comment (sophokles73) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: I know, but the demo apps are intended to be used from the command line (they all end up a single jar file apps in the *run* folder) and therefore IMHO they should support at least the bare minimum config without requiring users to recompile, don't you think?

File: demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java (hunk 1 above)
Comment (sophokles73) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: OutOfMemory. Have you made any good experience with catching it? My assumption would be that the system will already have become unstable when it occurs so there is not much point in catching it, I guess.

File: demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java (hunk 2 above)
Comment (boaks) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: OK, I add command line argument.

File: demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java (hunk 1 above)
Comment (boaks) [inline demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java]: OutOfMemory. Sure. In cases where "one big blob" is allocated, there is a small chance :-)

> byte[] content = new byte[(int) length]; 

But to keep it simple, I remove the OutOfMemory catch :-)

**Pull request merged**